src/test/java/com/google/sps/data/DatastoreClientTest.java

```diff
@@ -0,0 +1,155 @@
+package com.google.sps.data;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+//import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.Assert;
+import org.junit.Test;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import org.junit.After;
+import org.junit.Before;
+import java.util.Date;
+import java.util.Optional;
+import com.google.sps.data.Attendee;
+import com.google.sps.data.AttendeeInterface;
+import com.google.sps.data.Session;
+import com.google.sps.data.SessionInterface;
+import com.google.sps.data.Instance;
+import com.google.sps.data.InstanceInterface;
+import com.google.sps.data.DatastoreClient;
+import com.google.sps.data.DatastoreClientInterface;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/* Class that test the methods in DatastoreClient. */
+public class DatastoreClientTest  {
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testAttendeeIsInsertedToDatastore() {
+    // Tests that object is inserted to datastore.
+    DatastoreClientInterface datastoreClient = new DatastoreClient();
+    AttendeeInterface attendee = new Attendee("12345", "Taniece", new Date());
+    datastoreClient.insertOrUpdateAttendee(attendee);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Assert.assertEquals(datastore.prepare(new Query
+        (EntityConstants.AttendeeEntity.TABLE_NAME))
+        .countEntities(withLimit(10)), 1);
+    // Tests that getAttendee finds and returns attendee.
+    AttendeeInterface attendee2 =
+        datastoreClient.getAttendee("Taniece").get();
+    Assert.assertEquals(attendee, attendee2);
+    // Tests that existing attendees are updated after being modified.
+    datastoreClient.insertOrUpdateAttendee(attendee2);
+    Assert.assertEquals(datastore.prepare(new Query
+        (EntityConstants.AttendeeEntity.TABLE_NAME))
+        .countEntities(withLimit(10)), 1);
```

Comment:
Another case to test would be to add another new attendee and verify that we have 2 of them.

Reply:
This case is covered in testAttendeesInASession and testAvailableInstance for multiple attendee and instances entry. Will include a test for 2 sessions.